gap/PGMatrix.gi

```diff
@@ -1,142 +1,97 @@
-IsPGMatrix@ := function(pgMatrix)
-   local signature, quotient, PGMLst, item, itemLen;
-    
-    # check dimensions
-    if not Length(pgMatrix) = 3 then
-	return false;
-    fi;
-     
-    # get entries
-    signature := pgMatrix[1];
-    quotient := pgMatrix[2];
-    PGMLst := pgMatrix[3];
-
-    # check signature
-    if not IsTGSignature@(signature) then
-	return false;
-    fi;
+# Type PGMatrixElements
+DeclareRepresentation("IsPGMatrixElementsComponentRep", IsComponentObjectRep,
+	[ "signature", "quotient", "sparse", "elements" ]
```

Comment:
As I mentioned above, we might want to change the components to include the full and proper triangle groups instead of the signature, change the "quotient" component to a TGQuotient object and rename "elements" to "generators" or similar (besides potentially renaming PGMatrixElements itself).